### ISSUE TITLE
Add .npmignore

### ISSUE THREAD

**Issue by SimonSchick** [stripe/stripe-node#468]
I noticed you are shipping a large amount of files that are most likely not used by any consumers:

```
Misc:

node_modules/stripe/.eslintrc.js
node_modules/stripe/.travis.yml
node_modules/stripe/CHANGELOG.md
node_modules/stripe/.github/
node_modules/stripe/.github/ISSUE_TEMPLATE.md

Examples:

node_modules/stripe/examples/
node_modules/stripe/examples/webhook-signing/
node_modules/stripe/examples/webhook-signing/.eslintrc.js
node_modules/stripe/examples/webhook-signing/express.js
node_modules/stripe/examples/webhook-signing/package.json

Tests: 
node_modules/stripe/test/
node_modules/stripe/test/.eslintrc.js
node_modules/stripe/test/Error.spec.js
node_modules/stripe/test/StripeResource.spec.js
node_modules/stripe/test/Webhook.spec.js
node_modules/stripe/test/flows.spec.js
node_modules/stripe/test/mocha.opts
node_modules/stripe/test/stripe.spec.js
node_modules/stripe/test/testUtils.js
node_modules/stripe/test/utils.spec.js
node_modules/stripe/test/resources/
node_modules/stripe/test/resources/Account.spec.js
node_modules/stripe/test/resources/ApplePayDomains.spec.js
node_modules/stripe/test/resources/ApplicationFeeRefunds.spec.js
node_modules/stripe/test/resources/ApplicationFees.spec.js
node_modules/stripe/test/resources/Balance.spec.js
node_modules/stripe/test/resources/BitcoinReceivers.spec.js
node_modules/stripe/test/resources/ChargeRefunds.spec.js
node_modules/stripe/test/resources/Charges.spec.js
node_modules/stripe/test/resources/CountrySpecs.spec.js
node_modules/stripe/test/resources/Coupons.spec.js
node_modules/stripe/test/resources/CustomerCards.spec.js
node_modules/stripe/test/resources/CustomerSubscriptions.spec.js
node_modules/stripe/test/resources/Customers.spec.js
node_modules/stripe/test/resources/Disputes.spec.js
node_modules/stripe/test/resources/EphemeralKeys.spec.js
node_modules/stripe/test/resources/Events.spec.js
node_modules/stripe/test/resources/ExchangeRates.spec.js
node_modules/stripe/test/resources/FileUploads.spec.js
node_modules/stripe/test/resources/InvoiceItems.spec.js
node_modules/stripe/test/resources/Invoices.spec.js
node_modules/stripe/test/resources/IssuerFraudRecords.spec.js
node_modules/stripe/test/resources/LoginLinks.spec.js
node_modules/stripe/test/resources/OrderReturns.spec.js
node_modules/stripe/test/resources/Orders.spec.js
node_modules/stripe/test/resources/Payouts.spec.js
node_modules/stripe/test/resources/Plans.spec.js
node_modules/stripe/test/resources/Products.spec.js
node_modules/stripe/test/resources/RecipientCards.spec.js
node_modules/stripe/test/resources/Recipients.spec.js
node_modules/stripe/test/resources/Refunds.spec.js
node_modules/stripe/test/resources/SKUs.spec.js
node_modules/stripe/test/resources/Sources.spec.js
node_modules/stripe/test/resources/SubscriptionItems.spec.js
node_modules/stripe/test/resources/Subscriptions.spec.js
node_modules/stripe/test/resources/ThreeDSecure.spec.js
node_modules/stripe/test/resources/Tokens.spec.js
node_modules/stripe/test/resources/Topups.spec.js
node_modules/stripe/test/resources/TransferReversals.spec.js
node_modules/stripe/test/resources/Transfers.spec.js
node_modules/stripe/test/resources/UsageRecords.spec.js
node_modules/stripe/test/resources/data/
node_modules/stripe/test/resources/data/minimal.pdf
```

Adding these to the npmignore should significantly reduce the package size!

**Comment by brandur-stripe** [comment]
Yeah, this probably makes sense. I'll try to write one as a whitelist so that we can keep a fairly minimal set of files even as we add new ones.

**Comment by SimonSchick** [comment]
If you write it as a whitelist you can either use an invert glob, with the first line being `/*` and all subsequent ones starting with `!` OR you can use https://docs.npmjs.com/files/package.json#files

**Comment by brandur-stripe** [comment]
Excellent. Thanks!

**Comment by SimonSchick** [comment]
Please note that npm automatically adds essential files such as `LICENSE`, `README` and `package.json`, there is no need to include these files manually.
You can verify the packaged version using `npm pack` which generates a tgz file with all the contents that would be published to npm.

**Comment by brandur-stripe** [comment]
Released as 6.1.1.
